crates/sui-source-validation-service/src/main.rs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use sui_config::{sui_config_dir, SUI_CLIENT_CONFIG};
+use sui_sdk::wallet_context::WalletContext;
+use sui_source_validation_service::{initialize, serve};
+
+#[tokio::main]
+async fn main() -> anyhow::Result<()> {
+    let config = sui_config_dir()?.join(SUI_CLIENT_CONFIG);
+    let context = WalletContext::new(&config, None, None).await?;
+    let package_paths = vec![];
+    initialize(&context, package_paths).await?;
+    serve()?.await.map_err(anyhow::Error::from)
```

Comment:
nit: I would be surprised if you need the `anyhow::Error::from` here.

Reply:
I'm also a little bit confused by the set-up here is `HttpServer` a `Future`? Why don't we `await` it inside `serve`?

Reply:
>  is `HttpServer` a `Future`

yeah `HttpServer::new` creates a future. 

> `await` it inside `serve`?

We can, but we'll still need to have `serve` be `async` in that case, which means we still have to `await` in `main`.

I think this is ever slightly more clear so have applied the patch below.

> need the `anyhow::Error::from` here

The `HttpServer` future returns `Error` so need to map it. I could do something like `.map_err(|_| bail!("server didn't run blah blah")` instead--I think I understand that `annyhow::Error::from` doesn't look very traditional :-) Thoughts? Will follow up post merge...


```patch
--- a/crates/sui-source-validation-service/src/lib.rs
+++ b/crates/sui-source-validation-service/src/lib.rs
@@ -3,7 +3,7 @@
 
 use std::path::{Path, PathBuf};
 
-use actix_web::{dev::Server, web, App, HttpRequest, HttpServer, Responder};
+use actix_web::{web, App, HttpRequest, HttpServer, Responder};
 
 use move_package::BuildConfig as MoveBuildConfig;
 use sui_move::build::resolve_lock_file_path;
@@ -47,12 +47,12 @@ pub async fn initialize(
     Ok(())
 }
 
-pub fn serve() -> anyhow::Result<Server> {
-    Ok(
-        HttpServer::new(|| App::new().route("/api", web::get().to(api_route)))
-            .bind("0.0.0.0:8000")?
-            .run(),
-    )
+pub async fn serve() -> anyhow::Result<()> {
+    HttpServer::new(|| App::new().route("/api", web::get().to(api_route)))
+        .bind("0.0.0.0:8000")?
+        .run()
+        .await
+        .map_err(anyhow::Error::from)
 }
 
 async fn api_route(_request: HttpRequest) -> impl Responder {
diff --git a/crates/sui-source-validation-service/src/main.rs b/crates/sui-source-validation-service/src/main.rs
index 9cb357a24f..64cc5f3296 100644
--- a/crates/sui-source-validation-service/src/main.rs
+++ b/crates/sui-source-validation-service/src/main.rs
@@ -11,5 +11,5 @@ async fn main() -> anyhow::Result<()> {
     let context = WalletContext::new(&config, None, None).await?;
     let package_paths = vec![];
     initialize(&context, package_paths).await?;
-    serve()?.await.map_err(anyhow::Error::from)
+    serve().await
```

Reply:
Ah, you're right -- I was mis-attributing the `map_err` as applying to the return value of `serve` which is already an `anyhow::Error` when in fact it's for `HttpServer`.